src/Mappers/Generic.ts

```diff
@@ -110,7 +110,10 @@ export abstract class GenericMapper {
             return {kind: MatchResultKind.FAILED};
         }
         else if (GenericMapper.isMapLeaf(node)) {
-            const map = node as GenericMap;
+            // make a copy of the node object
+            const map = Object.assign({}, node) as GenericMap;
```

Comment:
This is a bug.
With `const map = node as GenericMap;`, `map` is a reference to `node` which is child of `this.root`.
Changes made to members inside `map` will apply to the referenced node in `this.root`, which causes only the latest state of `map` is used when `Mode.execute` is executed.

Thanks for the fix!